app/main.py

```
@@ -1,3 +1,65 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: int | float | "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+
+        if isinstance(other, (int, float)):
+            self.km += other
+        return self
```

Similar to `__add__`, this method silently does nothing if an unsupported type is provided. It should also `return NotImplemented` in such cases to ensure a `TypeError` is raised.